passes/pass4.nim

```diff
@@ -178,66 +178,73 @@ proc buildGraph(tree; n): Graph =
                dst: uint32(start + off),
                noCopy: tree[e].kind == Rename)
 
+        if tree[e].kind == Rename:
+          # for a 'rename' edge, mark both the source and destination as
+          # pinned
+          result.nodes[result.edges[^1].src].keep = true
+          result.nodes[result.edges[^1].dst].keep = true
+
     cont.groups.b = result.groups.high.int32
 
 proc colorGraph(gr: var Graph) =
-  ## Assigns a color to every node in the graph.
+  ## Assigns a color to every node in the graph. The goal is to use as little
```

Comment:
```suggestion
  ## Assigns a color to every node in the graph. The goal is to use as few
```

The intent is clear even without the change, but `little` is for non-countable things, while `few` is for countable.